evap/rewards/tools.py

```diff
@@ -68,10 +67,11 @@ def grant_reward_points(sender, **kwargs):
     if required_courses.exclude(voters=request.user).exists():
         return
     # did the user not already get reward points for this semester?
-    if not RewardPointGranting.objects.filter(user_profile=request.user, semester=semester):
-        granting = RewardPointGranting(user_profile=request.user, semester=semester, value=settings.REWARD_POINTS_PER_SEMESTER)
-        granting.save()
-        messages.success(request, _("You just have earned reward points for this semester because you evaluated all your courses. Thank you very much!"))
+    if RewardPointGranting.objects.filter(user_profile=request.user, semester=semester).exists():
```

Comment:
just checked, the method has 5 returns if a bunch of conditions is not met. i think it's nice to stay in that schema. also, the intent is expressed explicitly with this: if points were granted, do nothing, otherwise, grant points.